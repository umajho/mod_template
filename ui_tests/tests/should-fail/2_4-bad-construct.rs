fn main() {}

#[mod_template::define(define_foo; constructions(CONS -> ToCons))]
mod __ {
    #[__CONSTRUCT(CONS)]
    fn an_fn() {}
}

#[mod_template::define(define_foo; constructions(CONS))]
mod __ {
    #[__CONSTRUCT(foo as CONS)]
    fn an_fn() {}
}
