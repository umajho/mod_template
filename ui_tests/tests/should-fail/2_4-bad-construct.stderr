error: unexpected end of input, expected `as`
 --> tests/should-fail/2_4-bad-construct.rs:3:1
  |
3 | #[mod_template::define(define_foo; constructions(CONS -> ToCons))]
  | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  |
  = note: this error originates in the attribute macro `mod_template::define` (in Nightly builds, run with -Z macro-backtrace for more info)

error: expected `->`
 --> tests/should-fail/2_4-bad-construct.rs:9:54
  |
9 | #[mod_template::define(define_foo; constructions(CONS))]
  |                                                      ^
